docs/advanced.rst

```diff
@@ -64,13 +64,16 @@ That could be done in the spider's parser method:
     def parser(self, response: DummyRequest, page: MyPageObject):
         pass
 
-Spiders that annotate its first argument as ``DummyResponse`` are signaling that
-they're not going to make use of it, so it should be safe to skip it from our
-downloader middleware. This type annotation is already applied when you use the
-``callback_for`` helper.
+Spider method that has its first argument annotated as :class:`~.DummyResponse`
+is signaling that it is not going to use the response, so it should be safe
+to not download scrapy Response as usual.
 
-If neither spider callback or any of the input providers are going to make use
-of a ``Response``, our Injection Middleware skips download returning a
+This type annotation is already applied when you use the :func:`~.callback_for`
+helper: the callback which is created by ``callback_for`` doesn't use Response,
+it just calls page object's to_item method.
+
+If neither spider callback nor any of the input providers are using
+``Response``, InjectionMiddleware skips the download, returning a
```

Comment:
Same here: should we format this way?

```rst
``InjectionMiddleware``
```

Or even:

```rst
:class:`~.InjectionMiddleware`
```

Reply:
Added more links for InjectionMiddleware and DummyResponse, thanks!